qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -325,3 +334,63 @@ def iterations(self, value: List[int]) -> None:
             value: A new value for the powers.
         """
         self._iterations = value
+
+    @property
+    def circuit_results(self) -> Optional[Union[np.ndarray, Dict[str, int]]]:
+        """Return the circuit results. Can be a statevector or counts dictionary."""
+        return self._circuit_results
+
+    @circuit_results.setter
+    def circuit_results(self, value: Union[np.ndarray, Dict[str, int]]) -> None:
+        """Set the circuit results."""
+        self._circuit_results = value
+
+    @property
+    def top_measurement(self) -> str:
+        """Return the circuit results. Can be a statevector or counts dictionary."""
+        return self._top_measurement
+
+    @top_measurement.setter
+    def top_measurement(self, value: str) -> None:
+        """Set the circuit results."""
+        self._top_measurement = value
+
+    @property
+    def max_probability(self) -> float:
+        """Return the circuit results. Can be a statevector or counts dictionary."""
+        return self._max_probability
+
+    @max_probability.setter
+    def max_probability(self, value: float) -> None:
+        """Set the circuit results."""
+        self._max_probability = value
+
+    @property
+    def assignment(self) -> Callable[[str], Any]:
+        """Return the circuit results. Can be a statevector or counts dictionary."""
+        return self._assignment
+
+    @assignment.setter
+    def assignment(self, value: Callable[[str], Any]) -> None:
+        """Set the circuit results."""
+        self._assignment = value
+
+    @property
+    def oracle_evaluation(self) -> bool:
+        """Return the circuit results. Can be a statevector or counts dictionary."""
+        return self._oracle_evaluation
+
+    @oracle_evaluation.setter
+    def oracle_evaluation(self, value: bool) -> None:
+        """Set the circuit results."""
+        self._oracle_evaluation = value
+
+    @property
+    def shots(self) -> int:
```

Comment:
Since `shots` is set by the user on the QuantumInstance, do we adjust it where its necessary to provide this back. Is this something different or just some convenience to have it here in the result? (I see its taken here by summing the circuit_result values, if its not a statevector result, which is presumably something a user could do too if this were needed.

Reply:
Right, actually, I looked into 'AmplitudeAmplifierResult' for reference, and I found it has the `shots` property, so I copied it without thinking. However, after reading your comments, now I don't think that we need the `shots` property in here. We can get the information of shots from `QuantumInstance` (Also, having the setter for `shots` here sounds strange).